core/src/main/scala/com/pingcap/tispark/TiDBWriter.scala

```diff
@@ -0,0 +1,56 @@
+package com.pingcap.tispark
+
+import org.apache.spark.sql.{DataFrame, SQLContext, SaveMode, TiContext}
+
+object TiDBWriter {
+
+  def write(df: DataFrame,
+            sqlContext: SQLContext,
+            saveMode: SaveMode,
+            parameters: Map[String, String]): Unit = {
+    val options = new TiDBOptions(parameters)
+    val tiContext = new TiContext(sqlContext.sparkSession, Some(options))
+    val conn = TiDBUtils.createConnectionFactory(options)()
+
+    try {
+      val tableExists = TiDBUtils.tableExists(conn, options)
+      if (tableExists) {
+        saveMode match {
+          case SaveMode.Overwrite =>
+            if (options.isTruncate && TiDBUtils
+                  .isCascadingTruncateTable(options.url)
+                  .contains(false)) {
+              // In this case, we should truncate table and then load.
+              TiDBUtils.truncateTable(conn, options, tiContext)
+              val tableSchema = TiDBUtils.getSchemaOption(conn, options)
```

Comment:
can this move up to `if (tableExists)` ?

Reply:
no, in `Overwrite` mode, data must be delete first.